prog/vnet/subnet_nexus.rb

```diff
@@ -163,31 +163,24 @@ def gen_reqid
     end
   end
 
-  def self.random_private_ipv6(location)
+  def self.random_private_ipv6(location, project)
     network_address = NetAddr::IPv6.new((SecureRandom.bytes(7) + 0xfd.chr).unpack1("Q<") << 64)
     network_mask = NetAddr::Mask128.new(64)
     addr = NetAddr::IPv6Net.new(network_address, network_mask)
-    return random_private_ipv6(location) unless PrivateSubnet.where(net6: addr.to_s, location: location).first.nil?
+
+    addr = random_private_ipv6(location, project) if project.private_subnets.find { _1.net6.to_s == addr.to_s && _1.location == location }
```

Comment:
`project.private_subnets_dataset.where[Sequel[:net6] => addr.to_s, location: location]` or something like that?